devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -24,44 +26,64 @@ public AddExtensions(final File pom) throws IOException {
         this.pom = pom;
     }
 
-    public boolean addExtensions(final Set<String> extensions) throws IOException {
+    public boolean addExtensions(final Set<String> extensions) throws IOException, CommandsException {
         if (extensions == null || extensions.isEmpty()) {
             return false;
         }
 
         boolean updated = false;
         List<Dependency> dependenciesFromBom = getDependenciesFromBom();
 
+        List<Extension> loadedExtensions = MojoUtils.loadExtensions();
         for (String dependency : extensions) {
-            Optional<Extension> optional = MojoUtils.loadExtensions().stream()
-                    .filter(d -> {
-                        boolean hasTag = d.labels().contains(dependency.trim().toLowerCase());
-                        boolean machName = d.getName()
-                                .toLowerCase()
-                                .contains(dependency.trim().toLowerCase());
-                        return hasTag || machName;
-                    })
-                    .findAny();
-
-            if (optional.isPresent()) {
-                final Extension extension = optional.get();
-
-                if (!MojoUtils.hasDependency(model, extension.getGroupId(), extension.getArtifactId())) {
-                    System.out.println("Adding extension " + extension.managementKey());
-                    model.addDependency(extension
-                            .toDependency(containsBOM(model) &&
-                                    isDefinedInBom(dependenciesFromBom, extension)));
-                    updated = true;
+            Dependency res;
```

Comment:
I would do the `trim()` as a first step so that you don't have to do it elsewhere.